hyperspy/gui/egerton_quantification.py

```diff
@@ -369,10 +369,12 @@ def _interpolator_kind_changed(self, old, new):
         self.span_selector_changed()
 
     def _ss_left_value_changed(self, old, new):
-        self.span_selector_changed()
+        if not np.isnan(self.ss_right_value):
+            self.span_selector_changed()
 
     def _ss_right_value_changed(self, old, new):
-        self.span_selector_changed()
+        if not np.isnan(self.ss_right_value):
```

Comment:
This doesn't cause troubles because this better check is performed
[here](https://github.com/hyperspy/hyperspy/pull/581/files#r35655194).
However, I fully agree that it is better to do it properly here.

On Tue, 2015-07-28 at 07:43 -0700, vidartf wrote:
> In hyperspy/gui/egerton_quantification.py:
> >  
> >      def _ss_right_value_changed(self, old, new):
> > -        self.span_selector_changed()
> > +        if not np.isnan(self.ss_right_value):
> I still think you can get in problems when the values are being reset 
> (reset_span_selector()): When the first of the values is being set to 
> NaN, the other still has a valid value, so the event is triggered.
> I think it is best to check like this (as far as I can tell, that 
> shouldn't break anything that isn't already broken?):
> if not (np.isnan(self.ss_left_value) or 
> np.isnan(self.ss_right_value)):
> —
> Reply to this email directly or view it on GitHub.